src/Signer.php

```diff
@@ -19,16 +21,14 @@ public function algorithmId(): string;
      * Creates a hash for the given payload
      *
      * @throws CannotSignPayload  When payload signing fails.
-     * @throws InvalidKeyProvided When issue key is invalid/incompatible.
      * @throws ConversionFailed   When signature could not be converted.
      */
-    public function sign(string $payload, Key $key): string;
+    public function sign(string $payload): string;
```

Comment:
OpenSSL with encrypted keys requires us to either have a lot of `mixed` typehints or make Signers stateful.
I hate both, but preferred the latter to start this PR.